### PR TITLE
Issue 366: use Hash Set in the SecurityGroup. Also...

### PR DESCRIPTION
**Pull request by dmitryintel**
... remove the Mockito wrappers around the SecurityGroup in unit tests. https://github.com/opensecuritycontroller/osc-core/issues/366

### CODE REVIEW

**Comment by larkinscarvalho** [comment]
Please add reviewers to the PR

File: osc-server/src/main/java/org/osc/core/broker/service/tasks/conformance/openstack/deploymentspec/OsDAIConformanceCheckMetaTask.java [hunk 1]
```diff
@@ -191,6 +191,11 @@ public void executeTransaction(EntityManager em) throws Exception {
                 ApplianceSoftwareVersion currentSoftwareVersion = ds.getVirtualSystem().getApplianceSoftwareVersion();
                 boolean doesSvaVersionMatchVsVersion = false;
                 for (OsImageReference imageRef : ds.getVirtualSystem().getOsImageReference()) {
+                	
+                	if (sva.getImage() == null) {
```
Comment (arvindn05) [inline osc-server/src/main/java/org/osc/core/broker/service/tasks/conformance/openstack/deploymentspec/OsDAIConformanceCheckMetaTask.java]: unrelated change...please remove as its tracked by different bug

Comment (larkinscarvalho) [inline osc-server/src/main/java/org/osc/core/broker/service/tasks/conformance/openstack/deploymentspec/OsDAIConformanceCheckMetaTask.java]: How does this help? More details?

Comment (dmitryintel) [inline osc-server/src/main/java/org/osc/core/broker/service/tasks/conformance/openstack/deploymentspec/OsDAIConformanceCheckMetaTask.java]: I pushed changes relating to another bug. Inside that loop, I have seen the getImage to return null and this is the only sensible way to deal with the situation.

File: osc-domain/src/main/java/org/osc/core/broker/model/entities/virtualization/SecurityGroup.java [hunk 2]
```diff
@@ -75,7 +75,7 @@
 
     @OneToMany(mappedBy = "securityGroup", fetch = FetchType.LAZY, cascade = CascadeType.ALL)
     @OrderBy("type")
-    private Set<SecurityGroupMember> securityGroupMembers = new TreeSet<>();
+    private Set<SecurityGroupMember> securityGroupMembers = new HashSet<>();
```
Comment (arvindn05) [inline osc-domain/src/main/java/org/osc/core/broker/model/entities/virtualization/SecurityGroup.java]: one of the reasons why we used a tree set was to have a sorted list of members. Please check to see if any code relies on this collection being sorted.

Also since we are not putting the SecurityGroupMember in a tree set....it does not need to implement comparable interface. Please update the entity as required.

Comment (emanoelxavier) [inline osc-domain/src/main/java/org/osc/core/broker/model/entities/virtualization/SecurityGroup.java]: For entities coming from the db observe the OrderBy above. I believe the comparable wold still make sense if there is any piece of code that is initiating this field without db data, sorting it and expecting it would be by type. I am not sure if there is such case but if not I agree that the comparable can be removed.
